cli-quarkus/src/main/java/org/eclipse/hono/cli/adapter/amqp/AmqpAdapter.java

```diff
@@ -0,0 +1,614 @@
+                                msg.getReplyTo(),
+                                msg.getCorrelationId().toString(),
+                                responseStatusCode,
+                                Buffer.buffer("automatic response to [%s] command".formatted(msg.getSubject())),
+                                "text/plain",
+                                null))
+                    .onFailure(t -> {
+                        System.err.println("Could not send command response to Hono's AMQP adapter");
+                        CommandUtils.printError(t);
+                    }));
+        }
+    }
+
+    private String getConsumerKey(final String tenantId, final String deviceId) {
+        if (deviceId == null) {
+            return "@@@self@@@";
+        }
+        final var b = new StringBuilder();
+        b.append(deviceId);
+        if (tenantId != null) {
+            b.append("@").append(tenantId);
+        }
+        return b.toString();
+    }
+
+    @Override
+    public Integer call() {
+
+        try {
+            getClient()
+                .onSuccess(client -> connected.set(true))
+                .toCompletionStage()
+                .toCompletableFuture()
+                .join();
+        } catch (final CompletionException e) {
+            System.err.println("Failed to connect to Hono's AMQP adapter");
+            CommandUtils.printError(e.getCause());
+            return CommandLine.ExitCode.SOFTWARE;
+        }
+
+        readAndExecuteCommands();
+        return CommandLine.ExitCode.OK;
+    }
+
+    private void handleUnauthorizedError(final Throwable t, final String deviceId) {
+        if (ServiceInvocationException.extractStatusCode(t) == HttpURLConnection.HTTP_FORBIDDEN) {
+            System.err.println("""
+                    The currently connected device is not authorized to act on behalf of device [id: %1$s].
+                    In order to authorize the connected device, its device identifier needs to be added to
+                    the list of (gateway) devices that may act on behalf of device [id: %1$s].
+                    Please refer to https://www.eclipse.org/hono/docs/concepts/connecting-devices/#connecting-via-a-device-gateway
+                    for details regarding connecting devices via gateways.
+                    """.formatted(deviceId));
+        }
+    }
+
+    private void checkDeviceSpec(final String tenantId, final String deviceId) {
+        if (tenantId != null && deviceId == null) {
+            throw new ParameterException(
+                    spec.commandLine(),
+                    """
+                    Missing required option: '--device=<deviceId>'.
+                    """);
+        }
+    }
+
+    @CommandLine.Command(
+            name = "sub",
+            description = {"Start receiving commands for a device."},
+            mixinStandardHelpOptions = true,
+            subcommands = CommandLine.HelpCommand.class,
```

Comment:
I think it's a little strange to have "help" as the only sub-command here (seeing `sub [-hV] [-d=<arg1>] [-s=<arg2>] [-t=<arg0>] [COMMAND]` I wondered what kind of COMMAND could possibly be meant here) . I mean, to print help output you could already type `sub --help`. I would rather have expected that `help sub` could be used as an alternative, but that will actually send a telemetry message.

(And a minor inconsistency would also be, that the `consume` command doesn't have a `help` sub-command:
```
java -jar hono-cli-*-exec.jar app --host $AMQP_NETWORK_IP --port 15672 -u "consumer@HONO" -p "verysecret" --amqp consume help
Unmatched argument at index 11: 'help'
```
.)

Reply:
We could simply disable the *help* subcommand and just support the `-h` and `--help` options ...

Reply:
Yes, I would prefer that.

Reply:
I have already removed it ...